### PR TITLE
gh-115823: Calculate correctly error locations when dealing with implicit encodings

### PR DESCRIPTION
**Pull request by pablogsal**
* Issue: gh-115823

### CODE REVIEW

File: Parser/pegen_errors.c [hunk 1]
```diff
@@ -369,20 +369,19 @@ _PyPegen_raise_error_known_location(Parser *p, PyObject *errtype,
...
-            Py_ssize_t end_col_offset = _PyPegen_byte_offset_to_character_offset(error_line, end_col_number);
-            if (end_col_offset < 0) {
-                goto error;
-            } else {
-                end_col_number = end_col_offset;
-            }
+        } else {
+            end_col_number = end_col_offset;
         }
     }
```
Comment (lysnikolaou) [inline Parser/pegen_errors.c]: The way this is written is a bit confusing. Can we maybe rewrite it to something like this?

```c
    Py_ssize_t col_number, end_col_number = end_col_offset;

    col_number = _PyPegen_byte_offset_to_character_offset(error_line, col_offset);
    if (col_number < 0) {
        goto error;
    }

    if (end_col_offset > 0) {
        end_col_number = _PyPegen_byte_offset_to_character_offset(error_line, end_col_offset);
        if (end_col_number < 0) {
            goto error;
        }
    }
```

I think this is equivalent with the old version, and it avoids redeclaring `end_col_offset`.

Comment (pablogsal) [reply Parser/pegen_errors.c]: 👍 Makes sense, I will do the change.

Notice the diff looks a bit odd, but technically I am just removing the surrounding conditional.

Comment (lysnikolaou) [reply Parser/pegen_errors.c]: Yup, I know. My comment was more about the overall structure, not the change itself. Especially cause I found redeclaring `end_col_offset` a bit confusing while reviewing.

Comment (pablogsal) [reply Parser/pegen_errors.c]: Absolutely! I pushed a fix already, take a look :)

**Review by lysnikolaou — APPROVED** [review]
LGTM!

**Pull request merged**